reference/conanfile/attributes.rst

```diff
@@ -0,0 +1,1022 @@
+      - About the package_type and how it plays when a ``shared`` option is declared. <MISSING PAGE>
+
+
+
+.. _conan_conanfile_properties_default_options:
+
+default_options
+---------------
+
+.. autoattribute:: ConanFile.default_options
+
+
+   .. code-block:: python
+
+       class MyPkg(ConanFile):
+           ...
+           requires = "zlib/1.2.8", "zwave/2.0"
+           options = {"build_tests": [True, False],
+                      "option2": "ANY"}
+           default_options = {"build_tests": True,
+                              "option1": 42,
+                              "z*: shared": True}
+
+
+   You can also assign default values for options of your requirements using "<reference_pattern>: option_name", being
+   a valid ``reference_pattern`` a ``name/version`` or any pattern with ``*`` like the example above.
+
+   You can also set the options conditionally to a final value with ``configure()`` instead of using ``default_options``:
+
+   .. code-block:: python
+
+       class OtherPkg(ConanFile):
+           settings = "os", "arch", "compiler", "build_type"
+           options = {"some_option": [True, False]}
+           # Do NOT declare 'default_options', use 'config_options()'
+
+           def configure(self):
+               if self.options.some_option == None:
+                   if self.settings.os == 'Android':
+                       self.options.some_option = True
+                   else:
+                       self.options.some_option = False
+
+   Take into account that if a value is assigned in the ``configure()`` method it cannot be overridden.
+
+
+   .. seealso::
+
+       Read more about the <MISSING PAGE>method_configure_config_options method.
+
+
+options_description
+-------------------
+
+TODO: Complete, https://github.com/conan-io/conan/pull/11295
+
+
+requires
+--------
+
+.. autoattribute:: ConanFile.requires
+
+
+    .. code-block:: python
+
+        class MyLibConan(ConanFile):
+            requires = "hello/1.0", "OtherLib/2.1@otheruser/testing"
+
+
+  You can specify version ranges, the syntax is using brackets:
+
+    .. _version_ranges_reference:
+
+
+    ..  code-block:: python
+
+        class HelloConan(ConanFile):
+            requires = "pkg/[>1.0 <1.8]"
+
+
+  Accepted expressions would be:
+
+    ..  code-block:: python
+
+          >1.1 <2.1    # In such range
+          2.8          # equivalent to =2.8
+          ~=3.0        # compatible, according to semver
+          >1.1 || 0.8  # conditions can be OR'ed
+
+
+  .. seealso::
+
+       - Check <MISSING PAGE> version_ranges if you want to learn more about version ranges.
+       - Check <MISSING PAGE> requires() conanfile.py method.
+
+tool_requires
+--------------
+
+.. autoattribute:: ConanFile.tool_requires
+
+
+  .. code-block:: python
+
+     class MyPkg(ConanFile):
+         tool_requires = "tool_a/0.2", "tool_b/0.2@user/testing"
+
+
+  .. seealso::
+
+      - Check <MISSING PAGE> tool_requires() conanfile.py method.
```

Comment:
I think I would replace this with a paragraph explaining that this is a declarative way of adding the tool_requires but if you want to do that "programmatically" you have the tool_requires() method and add the link. The same for build_requires, etc...